### PR TITLE
OkHttpClientHelperTest: Add a test for getting different clients

### PR DESCRIPTION
**Pull request by sschuberth**
Signed-off-by: Sebastian Schuberth <sschuberth@gmail.com>

### CODE REVIEW

File: utils/ort/src/test/kotlin/OkHttpClientHelperTest.kt [hunk 1]
```diff
@@ -58,6 +59,24 @@ class OkHttpClientHelperTest : WordSpec({
...
+        "return different clients when specifying different configurations" {
+            val timeoutA: BuilderConfiguration = { readTimeout(Duration.ofSeconds(100)) }
+            val timeoutB: BuilderConfiguration = { readTimeout(Duration.ofSeconds(500)) }
+            val clientA = OkHttpClientHelper.buildClient(timeoutA)
+            val clientB = OkHttpClientHelper.buildClient(timeoutB)
+
+            clientA shouldNotBeSameInstanceAs clientB
+        }
+
+        "return different clients when specifying different configurations instances with the same values" {
```
Comment (fviernau) [inline utils/ort/src/test/kotlin/OkHttpClientHelperTest.kt:72]: configurations instances -> configuration instances?

File: utils/ort/src/test/kotlin/OkHttpClientHelperTest.kt [hunk 2]
```diff
@@ -58,6 +59,24 @@ class OkHttpClientHelperTest : WordSpec({
 
             clientA shouldBeSameInstanceAs clientB
         }
+
+        "return different clients when specifying different configurations" {
```
Comment (fviernau) [inline utils/ort/src/test/kotlin/OkHttpClientHelperTest.kt:63]: Would it make sense to say here: different configuration instances with same values?

(To emphasize that it is just about instance equality, not about the equality of the values it sets).

**Review by fviernau — CHANGES_REQUESTED** [review]

File: utils/ort/src/test/kotlin/OkHttpClientHelperTest.kt (hunk 2 above)
Comment (sschuberth) [reply utils/ort/src/test/kotlin/OkHttpClientHelperTest.kt:63]: > different configuration instances with same values

But that's the test case below, isn't it?

File: utils/ort/src/test/kotlin/OkHttpClientHelperTest.kt (hunk 1 above)
Comment (sschuberth) [reply utils/ort/src/test/kotlin/OkHttpClientHelperTest.kt]: I've reworded this yet more to emphasize that differently clients are being returned despite the configuration being the same (value-wise).

File: utils/ort/src/test/kotlin/OkHttpClientHelperTest.kt (hunk 2 above)
Comment (fviernau) [reply utils/ort/src/test/kotlin/OkHttpClientHelperTest.kt:63]: The proposal was for the one in line 63 - anyhow, leaving it up to you.

**Review by fviernau — APPROVED** [review]

File: utils/ort/src/test/kotlin/OkHttpClientHelperTest.kt (hunk 2 above)
Comment (sschuberth) [reply utils/ort/src/test/kotlin/OkHttpClientHelperTest.kt:63]: > The proposal was for the one in line 63 - anyhow, leaving it up to you.

What I was trying to say is that IMO it's incorrect to rename the test in line 63 to "different configuration instances with same values" as that's a different case that's already tested in line 72.

**Pull request merged**